### PR TITLE
mockserver locking, vsock address handling and CI

### PR DESCRIPTION
**Pull request by bergwolf**
1. enable mock locking so that parallel client RPC does not cause data race.
2. fix vsock address handling error due to url.URL vs. gRPC address format difference.
3. enable travis ci for kata agent.
4. add vendor so that travis can run properly.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#
+
+sudo: required
+dist: trusty
+
+language: go
+go_import_path: github.com/kata-containers/agent
+
+go:
+  - 1.8
```
Comment (jodh-intel) [inline .travis.yml]: How about also adding the following?:

```
  - 1.9
  - tip
```

Comment (bergwolf) [reply .travis.yml]: I'm OK with both. Just note that other components are also using v1.8. If we want to go with v1.9 and tip, we should change others' travis settings as well.

Comment (jodh-intel) [reply .travis.yml]: Ah - sorry, I forgot to check the other components. I'm happy to raise PRs to add these 2 versions in for the others if we all agree to use `1.9` and `tip` though?

Comment (bergwolf) [reply .travis.yml]: yeah, sounds good to me!

**Comment by laijs** [comment]
LGTM

**Comment by laijs** [comment]
@bergwolf Could you cherry-pick the #34 and update the commit. Please ensure the author name unchanged.

**Comment by bergwolf** [comment]
cherry-picked #34 as the last commit and CI passed.

**Pull request merged**